### PR TITLE
bpo-38994: Implement __class_getitem__ for PathLike

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue38994

Automerge-Triggered-By: @asvetlov

### CODE REVIEW

**Review by asvetlov — CHANGES_REQUESTED** [review]
Adding `__class_getitem__` to `os.PathLike` only is not enough for using, e.g. `pathlib.Path[str]` in runtime.

`pathlib.PurePath` should be modified in this way as well.

**Comment by isidentical** [comment]
@asvetlov I have made the requested changes; please review again

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
Awesome, thanks!

File: Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Implement ``__class_getitem__`` for ``os.PathLike``, ``pathlib.Path``
```
Comment (brettcannon) [inline Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst:1]: This news item is slightly misleading as the change was made to `pathlib.PurePath`, not `pathlib.Path`.

Comment (brettcannon) [reply Misc/NEWS.d/next/Library/2019-12-07-18-58-44.bpo-38994.IJYhz_.rst:1]: And @isidentical forgot to give himself credit for the change. :)

**Comment by brettcannon** [comment]
Any reason not to document this?

**Comment by isidentical** [comment]
It is kind of irrelevant I wrote a simple script to analyze typeshed and I think i found other classes with missing class getitems. Should I open an issue per class or just open one and fix all of them in one PR (I can tweak the comment change @brettcannon said in that PR). It is like 6-7 lines per class so i dont think it would make review harder.

**Comment by brettcannon** [comment]
@isidentical individual is always easier to review, but if it's too much work you can try doing it in a single PR.

**Comment by asvetlov** [comment]
@isidentical would you make a follow-up PR for fixing @brettcannon notes?

**Comment by isidentical** [comment]
Yes

**Pull request merged**